src/Compilers/Lua/Test/Portable/Lexical/RegressionTests.cs

```diff
@@ -143,4 +143,31 @@ await Assert.That(token).HasKind(SyntaxKind.StringLiteralToken).And.HasText(rawT
             token.GetDiagnostics().Verify();
         }
     }
+    
+    [Test]
+    [WorkItem(152, "https://github.com/LorettaDevs/Loretta/issues/152")]
+    public Task Lexer_TokenCacheCorrectlyHandlesSyntaxOptions()
+    {
+        var str = @"
+                     repeat
+                         m, E = G:zR(E, H);
+                         if m == 0x7423 then
+                             break;
+                         else
+                             if m ~= 0X76D4 then
+                             else
+                                 continue;
+                             end;
+                         end;
+                     until false;
+        ";
+        
+        var syntaxTree = LuaSyntaxTree.ParseText(str, new LuaParseOptions(LuaSyntaxOptions.Lua51));
+        var syntaxTree2 = LuaSyntaxTree.ParseText(str, new LuaParseOptions(LuaSyntaxOptions.Luau));
+        
+        if (syntaxTree.GetDiagnostics().Any() && syntaxTree2.GetDiagnostics().Any())
+            Assert.Fail("Token cache did not handle syntax options correctly");
```

Comment:
Instead of checking if there are any diagnostics, check that they don't share the same diagnostics instead.

Preferably use `.Verify`.